gpbft/rebroadcast.go

```diff
@@ -0,0 +1,151 @@
+package gpbft
+
+import (
+	"fmt"
+	"time"
+)
+
+type rebroadcastState struct {
+	timeout  time.Time
+	attempts int
+
+	decideCurrentRound,
+	commitCurrentRound,
+	commitPreviousRound,
+	prepareCurrentRound,
+	preparePreviousRound,
+	convergeCurrentRound *MessageBuilder
+
+	instanceAtLatestBroadcast uint64
```

Comment:
This is all in the context of a single GPBFT instance, so this instance can never change.